src/main/java/duke/command/Duke.java

```diff
@@ -0,0 +1,277 @@
+            return;
+        }
+
+        switch (taskType) {
+        case TODO:
+            taskList.add((new ToDo(taskDescription)));
+            break;
+        case DEADLINE:
+            if (!taskDescription.contains("/by")) {
+                printDividerLine();
+                System.out.println("Missing /by time!");
+                printDividerLine();
+                return;
+            }
+            taskList.add((new Deadline(taskDescription)));
+            break;
+        case EVENT:
+            if (!taskDescription.contains("/at")) {
+                printDividerLine();
+                System.out.println("Missing /at time!");
+                printDividerLine();
+                return;
+            }
+            taskList.add((new Event(taskDescription)));
+            break;
+        }
+
+        printDividerLine();
+        System.out.println("I have added this task:");
+        System.out.println("[" + taskList.get(numberOfTasks).getType() + "][" + taskList.get(numberOfTasks).getStatusIcon() + "] " + taskList.get(numberOfTasks).getDescription());
+        numberOfTasks++;
+        System.out.println("You have " + numberOfTasks + " task(s) in the list.");
+        printDividerLine();
+
+        try {
+            writeToFile();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    private static void displayList() {
+        printDividerLine();
+        if (numberOfTasks == 0) {
+            System.out.println("to-do list is empty! add something");
+            return;
+        }
+        System.out.println("The current to-do list is as follows:");
+        int counter = 1;
+        for (Task item : taskList) {
+            System.out.println(counter + ". [" + item.getType() + "][" + item.getStatusIcon() + "] " + item.getDescription());
+            counter++;
+        }
+        printDividerLine();
+    }
+
+    private static void markTaskComplete(String taskDescription) {
+        int taskNumber;
+
+        try {
+            taskNumber = Integer.parseInt(taskDescription);
+            if (taskNumber > numberOfTasks || taskNumber <= 0) {
+                System.out.println("Error! This task does not exist!");
+                return;
+            }
+            taskNumber--;
+            taskList.get(taskNumber).markAsDone();
+            printDividerLine();
+            System.out.println("I have marked it as completed!");
+            System.out.println(taskNumber + 1 + ". [" + taskList.get(taskNumber).getType() + "][" + taskList.get(taskNumber).getStatusIcon() + "] " + taskList.get(taskNumber).getDescription());
+            printDividerLine();
+        } catch (NumberFormatException e) {
+            System.out.println("Error! This task does not exist!");
+        }
+        try {
+            writeToFile();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
```

Comment:
This is a second `try-catch` block in the method. You could actually catch multiple exceptions following a single `try`.